Closes #1195 - Fix copy constructors to ensure value-based copy

Omar-Ballesteros opened the pull request:
Closes #1195 

This PR fixes and standardizes several copy constructors as part of #1195 
- Ensures fields are copied by value where needed.
- Leaves entity-type fields copied by reference, as confirmed in the issue discussion.
- Adds any missing fields that were previously not copied.

Updated classes
- `TaskImpl`, `TaskSummaryImpl`, `TaskCommentImpl`

- `AttachmentSummaryImpl`, `ObjectReferenceImpl`

- `WorkbasketImpl`, `WorkbasketSummaryImpl`

- `ClassificationSummaryImpl`

- `UserSummaryImpl`

Other classes like `UserImpl`, `WorkbasketAccessItemImpl`, and `ClassificationImpl` were reviewed but required no changes.
<!-- if needed please write above the given line -->

## Definition of Done

- [x] The corresponding ticket is in state `In Review`
- [x] The corresponding ticket is attached to this Pull-Request
- [x] The commit-message-format `Closes #ISSUE_ID - PROBLEM/SOLUTION` 
  - is present as single-commit already or
  - will be squashed on merge
- [ ] The changes pass the SonarQubeCloud-Quality-Gate
- [x] If the documentation needs an update, following there's a link to it:
- [x] If extra release-notes are required, they're listed indented below:

Omar-Ballesteros committed:
Closes #1195 - Fix copy constructors to ensure value-based copy

bruderj15 commented on lib/kadai-core/src/main/java/io/kadai/user/internal/models/UserSummaryImpl.java:
```diff
@@ -46,9 +47,8 @@ public class UserSummaryImpl implements UserSummary {
   public UserSummaryImpl() {}
 
   protected UserSummaryImpl(UserSummaryImpl copyFrom) {
```
persistence-fields are to be ignored

bruderj15 commented on lib/kadai-core/src/main/java/io/kadai/classification/internal/models/ClassificationSummaryImpl.java:
```diff
@@ -66,6 +66,7 @@ protected ClassificationSummaryImpl(ClassificationSummaryImpl copyFrom) {
     custom6 = copyFrom.custom6;
     custom7 = copyFrom.custom7;
     custom8 = copyFrom.custom8;
+    key = copyFrom.key;
```
Can you move this assignement a few rows up, below `domain`, such that we keep it in sync with the order of the class attributes?

bruderj15 commented on lib/kadai-core/src/main/java/io/kadai/task/internal/models/TaskImpl.java:
```diff
@@ -48,11 +48,13 @@ public TaskImpl() {}
+    this.attachments = copyFrom.attachments
+            .stream()
+            .map(Attachment::copy)
+            .collect(Collectors.toList());
```
Copilot is right!

Omar-Ballesteros committed:
Fix: Restore removed line and aligned assignment order

Omar-Ballesteros commented:
Thanks for the feedback! I've addressed both comments: restored the removed line and moved the assignment as requested. Let me know if there's anything else I should adjust.

bruderj15 reviewed (approved):
Great job!
